File: web/tests/reference/graphs/number-of-provinces.js
function findCircleNum(isConnected) {
  const n = isConnected.length;
  const seen = new Array(n).fill(false);
  let count = 0;
  for (let start = 0; start < n; start++) {
    if (seen[start]) continue;
    count += 1;
    const stack = [start];
    seen[start] = true;
    while (stack.length > 0) {
      const node = stack.pop();
      for (let j = 0; j < n; j++) {
        if (isConnected[node][j] === 1 && !seen[j]) {
          seen[j] = true;
          stack.push(j);
        }
      }
    }
  }
  return count;
}

File: web/tests/reference/graphs/number-of-provinces.py
def find_circle_num(is_connected):
    n = len(is_connected)
    seen = [False] * n
    count = 0
    for start in range(n):
        if seen[start]:
            continue
        count += 1
        stack = [start]
        seen[start] = True
        while stack:
            node = stack.pop()
            for j in range(n):
                if is_connected[node][j] == 1 and not seen[j]:
                    seen[j] = True
                    stack.append(j)
    return count
